moveit_ros/moveit_servo/src/servo_calcs.cpp

```diff
@@ -989,6 +989,11 @@ Eigen::VectorXd ServoCalcs::scaleCartesianCommand(const geometry_msgs::msg::Twis
   Eigen::VectorXd result(6);
   result.setZero();  // Or the else case below leads to misery
 
+  // Add a user-defined, constant delay to the timestep.
+  // This can help if ramp-up / ramp-down of the low-level controllers cause jitter, or to account for network latency.
+  // Effectively it moves the target pose farther ahead.
+  double timestep = parameters_->publish_period + parameters_->target_pose_lookahead_time;
```

Comment:
Where's the `timestep` being used? I can be marked as `const`.

Reply:
I agree with @jliukkonen.
Instead of [this block](https://github.com/AndyZe/moveit2/blob/7fef7faefdac357da44ff01ceaa15838919a7ae7/moveit_ros/moveit_servo/src/servo_calcs.cpp#L1034-L1054):
Consider this:
```
  const double timestep = parameters_->publish_period + parameters_->target_pose_lookahead_time;

  std::size_t c;
  for (std::size_t m = 0; m < command.joint_names.size(); ++m)
  {
    try
    {
      c = joint_state_name_map_.at(command.joint_names[m]);
    }
    catch (const std::out_of_range& e)
    {
      rclcpp::Clock& clock = *node_->get_clock();
      RCLCPP_WARN_STREAM_THROTTLE(LOGGER, clock, ROS_LOG_THROTTLE_PERIOD, "Ignoring joint " << command.joint_names[m]);
      continue;
    }
    // Apply user-defined scaling if inputs are unitless [-1:1]
    if (parameters_->command_in_type == "unitless")
      result[c] = command.velocities[m] * parameters_->joint_scale * timestep;
    // Otherwise, commands are in m/s and rad/s
    else if (parameters_->command_in_type == "speed_units")
      result[c] = command.velocities[m] *timestep;
```

And instead of [this block](https://github.com/AndyZe/moveit2/blob/7fef7faefdac357da44ff01ceaa15838919a7ae7/moveit_ros/moveit_servo/src/servo_calcs.cpp#L995-L1015)

Consider using following:
```
  const double timestep = parameters_->publish_period + parameters_->target_pose_lookahead_time;

  // Apply user-defined scaling if inputs are unitless [-1:1]
  if (parameters_->command_in_type == "unitless")
  {
    result[0] = parameters_->linear_scale *timestep * command.twist.linear.x;
    result[1] = parameters_->linear_scale * timestep * command.twist.linear.y;
    result[2] = parameters_->linear_scale * timestep * command.twist.linear.z;
    result[3] = parameters_->rotational_scale * timestep * command.twist.angular.x;
    result[4] = parameters_->rotational_scale *timestep * command.twist.angular.y;
    result[5] = parameters_->rotational_scale * timestep * command.twist.angular.z;
  }
  // Otherwise, commands are in m/s and rad/s
  else if (parameters_->command_in_type == "speed_units")
  {
    result[0] = command.twist.linear.x * timestep;
    result[1] = command.twist.linear.y * timestep;
    result[2] = command.twist.linear.z *timestep;
    result[3] = command.twist.angular.x * timestep;
    result[4] = command.twist.angular.y * timestep;
    result[5] = command.twist.angular.z *timestep;